crypto/src/asset.rs

```diff
@@ -21,7 +23,7 @@ pub use registry::{Registry, REGISTRY};
 #[serde(try_from = "pb::Asset", into = "pb::Asset")]
 pub struct Asset {
     pub id: Id,
-    pub denom: Denom,
+    pub denom: DenomMetadata,
 }
```

Comment:
What's the point of the `Asset` proto message, now that we have `DenomMetadata` ?

Reply:
Hmm, yes, seems redundant. Should I add removing `Asset` and replacing with `DenomMetadata` to this refactor?

Reply:
I don't know. This PR is already terrifyingly large, maybe we should try to land it and then do the `Asset` removal in another step?

Reply:
I was hoping you would suggest that :) sounds great